d3a_interface/utils.py

```diff
@@ -132,5 +134,18 @@ def mkdir_from_str(directory: str, exist_ok=True, parents=True):
 class RepeatingTimer(Timer):
     def run(self):
         while not self.finished.is_set():
-            self.function(*self.args, **self.kwargs)
             self.finished.wait(self.interval)
+            self.function(*self.args, **self.kwargs)
```

Comment:
Is it really necessary to change the order here? I am not negative on that, however it would be great if you checked the simulation results via the UI after this change, to validate that the GraphQL subscriptions (which also use this timer) work as expected

Reply:
Remnant of testing, reverting it